sdk/core/System.ClientModel/tests/client/ModelReaderWriter/ResourceDataWriteProxy.cs

```diff
@@ -0,0 +1,129 @@
+            {
+                if (property.NameEquals("id"u8))
+                {
+                    continue;
+                }
+                if (property.NameEquals("namespace"u8))
+                {
+                    @namespace = property.Value.GetString();
+                    continue;
+                }
+                if (property.NameEquals("registrationState"u8))
+                {
+                    registrationState = property.Value.GetString();
+                    continue;
+                }
+                if (property.NameEquals("registrationPolicy"u8))
+                {
+                    registrationPolicy = property.Value.GetString();
+                    continue;
+                }
+                if (property.NameEquals("resourceTypes"u8))
+                {
+                    if (property.Value.ValueKind == JsonValueKind.Null)
+                    {
+                        continue;
+                    }
+                    List<ProviderResourceType> array = new List<ProviderResourceType>();
+                    foreach (var item in property.Value.EnumerateArray())
+                    {
+                        array.Add(ProviderResourceType.DeserializeProviderResourceType(item, options));
+                    }
+                    resourceTypes = array;
+                    continue;
+                }
+                if (property.NameEquals("providerAuthorizationConsentState"u8))
+                {
+                    if (property.Value.ValueKind == JsonValueKind.Null)
+                    {
+                        continue;
+                    }
+                    providerAuthorizationConsentState = new ProviderAuthorizationConsentState(property.Value.GetString());
+                    continue;
+                }
+            }
+            return new ResourceProviderData(id.Value, @namespace.Value, registrationState.Value, registrationPolicy.Value, OptionalProperty.ToList(resourceTypes), OptionalProperty.ToNullable(providerAuthorizationConsentState));
+        }
+
+        ResourceProviderData IJsonModel<ResourceProviderData>.Create(ref Utf8JsonReader reader, ModelReaderWriterOptions options)
+        {
+            using var doc = JsonDocument.ParseValue(ref reader);
+            return DeserializeResourceProviderData(doc.RootElement, options);
+        }
+
+        ResourceProviderData IPersistableModel<ResourceProviderData>.Create(BinaryData data, ModelReaderWriterOptions options)
+        {
+            using var doc = JsonDocument.Parse(data);
+            return DeserializeResourceProviderData(doc.RootElement, options);
+        }
+
+        string IPersistableModel<ResourceProviderData>.GetFormatFromOptions(ModelReaderWriterOptions options) => "J";
+
+        void IJsonModel<ResourceProviderData>.Write(Utf8JsonWriter writer, ModelReaderWriterOptions options) => Write((ResourceProviderData)options.ProxiedModel!, writer, options);
```

Comment:
How do we guarantee that null suppression is safe here?  What about the MRW contract ensures that the options passed will always have this populated?